nengo/networks/assoc_mem.py

```diff
@@ -94,49 +102,139 @@ def __init__(self, input_vectors, output_vectors=None,  # noqa: C901
                 attr='threshold', obj=self)
 
         # --- Set parameters
-        self.out_conns = []  # Used in `add_threshold_to_output`
-        # Used in `add_threshold_to_output`
-        self.default_vector_inhibit_conns = []
-        self.thresh_ens = None  # Will hold thresholded outputs
         self.is_wta = False
         self._inhib_scale = 1.5
+        self._output_vectors = {}
+        self._default_output_vectors = {}
 
         # -- Create the core network
         with self, self.am_ens_config:
             self.bias_node = nengo.Node(output=1)
             self.elem_input = nengo.Node(
                 size_in=self.n_items, label="element input")
-            self.elem_output = nengo.Node(
-                size_in=self.n_items, label="element output")
-            self.utilities = self.elem_output
+            self.elem_utilities = nengo.Node(
```

Comment:
True, but if I have to go to this code later, I'd like to know what
organisation you had in mind.

On Tue, May 3, 2016 at 1:55 PM, xchoo <notifications@github.com> wrote:

> In nengo/networks/assoc_mem.py
> <https://github.com/nengo/nengo/pull/991#discussion_r61925746>:
>
> >
> >          # -- Create the core network
> >          with self, self.am_ens_config:
> >              self.bias_node = nengo.Node(output=1)
> >              self.elem_input = nengo.Node(
> >                  size_in=self.n_items, label="element input")
> > -            self.elem_output = nengo.Node(
> > -                size_in=self.n_items, label="element output")
> > -            self.utilities = self.elem_output
> > +            self.elem_utilities = nengo.Node(
>
> The 'elements' aren't really meant to be outward facing. Users really
> don't need to know what an 'element' in the code corresponds to. All they
> need are the input and output nodes.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/nengo/nengo/pull/991/files/c26689d6a5768984179442bc21b3cde8d46a1580#r61925746>
>